src/main/java/jnr/unixsocket/DgramUnixSocketChannel.java

```diff
@@ -0,0 +1,281 @@
+    return new DgramUnixSocketChannel();
+  }
+
+  /**
+   * connect to an unixSocketAddress.
+   *
+   * @param remote unixSocketAddress
+   * @return this
+   * @throws IOException if socket fails to connect
+   */
+  public static final DgramUnixSocketChannel open(UnixSocketAddress remote) throws IOException {
+    DgramUnixSocketChannel channel = new DgramUnixSocketChannel();
+    channel.connect(remote);
+    return channel;
+  }
+
+  public static final OutputStream openOutputStream(UnixSocketAddress remote) throws IOException {
+    DgramUnixSocketChannel channel = new DgramUnixSocketChannel();
+    channel.connect(remote);
+    return Channels.newOutputStream(channel);
+  }
+
+  /**
+   * invoke native socketpair.
+   *
+   * @return an array of two DgramUnixSocketChannels
+   * @throws IOException if socket fails to connect
+   */
+  public static final DgramUnixSocketChannel[] pair() throws IOException {
+    int[] sockets = {-1, -1};
+    Native.socketpair(ProtocolFamily.PF_UNIX, Sock.SOCK_DGRAM, 0, sockets);
+    return new DgramUnixSocketChannel[]{
+        new DgramUnixSocketChannel(sockets[0], SelectionKey.OP_READ | SelectionKey.OP_WRITE),
+        new DgramUnixSocketChannel(sockets[1], SelectionKey.OP_READ | SelectionKey.OP_WRITE)
+    };
+  }
+
+  private DgramUnixSocketChannel() throws IOException {
+    super(Native.socket(ProtocolFamily.PF_UNIX, Sock.SOCK_DGRAM, 0),
+        SelectionKey.OP_CONNECT | SelectionKey.OP_READ | SelectionKey.OP_WRITE);
+    state = State.IDLE;
+  }
+
+  DgramUnixSocketChannel(int fd, int ops) {
+    super(fd, ops);
+    state = State.CONNECTED;
+  }
+
+  DgramUnixSocketChannel(int fd, UnixSocketAddress remote) {
+    super(fd, SelectionKey.OP_READ | SelectionKey.OP_WRITE);
+    state = State.CONNECTED;
+    remoteAddress = remote;
+  }
+
+  private boolean doConnect(SockAddrUnix remote) throws IOException {
+    if (Native.connect(getFD(), remote, remote.length()) != 0) {
+      Errno error = Errno.valueOf(LastError.getLastError(jnr.ffi.Runtime.getSystemRuntime()));
+
+      switch (error) {
+        case EAGAIN:
+        case EWOULDBLOCK:
+          return false;
+
+        default:
+          throw new IOException(error.toString());
+      }
+    }
+    return true;
+  }
+
+  /**
+   * check whether remote unix socket is connected.
+   *
+   * @param remote remote unix socket address
+   * @return whether remote unix socket is connected
+   * @throws IOException if socket fails to connect
+   */
+  public boolean connect(UnixSocketAddress remote) throws IOException {
+    remoteAddress = remote;
+    if (!doConnect(remoteAddress.getStruct())) {
+
+      state = State.CONNECTING;
+      return false;
+
+    } else {
+
+      state = State.CONNECTED;
+      return true;
+    }
+  }
+
+  /**
+   * check whether current socket is connected.
+   *
+   * @return whether current connection is connected
+   */
+  public boolean isConnected() {
```

Comment:
They should probably just return sensible defaults for a dgram (presumably connectionless) socket.